gauntlet/packages/gauntlet-solana-contracts/src/lib/contracts.ts

```diff
@@ -39,18 +39,25 @@ export const getContract = (name: CONTRACT_LIST, version: string): Contract => (
   programId: getProgramId(name),
 })
 
-export const getDeploymentContract = (name: CONTRACT_LIST, version: string): DeploymentContract => ({
+export const getDeploymentContract = async (name: CONTRACT_LIST, version: string): Promise<DeploymentContract> => ({
   id: name,
   programKeypair: getProgramKeypair(name, version),
-  bytecode: getContractCode(name, version),
+  bytecode: await getContractCode(name, version),
 })
 
-// TODO: Get it from GH Releases
-const getContractCode = (name: CONTRACT_LIST, version: string) => {
+const getContractCode = async (name: CONTRACT_LIST, version: string): Promise<Buffer> => {
   try {
-    return readFileSync(join(process.cwd(), 'packages/gauntlet-solana-contracts/artifacts/bin', `${name}.so`))
+    if (version === 'local') {
+      return readFileSync(join(process.cwd(), 'packages/gauntlet-solana-contracts/artifacts/bin', `${name}.so`))
```

Comment:
@RodrigoAD should we rather point to the artifact source from here which gets built under `./contracts/artifacts/...`?

Reply:
We don't track the schemas artifacts in `./contracts`

Reply:
We'll need a solution to handle binary & schema loading both for our local code and vendor (multisig).

Reply:
If the IDL is stored on chain then we don't need to have a local copy, `anchor.Program.at(program_id)` is enough to construct an instance. But then we must ensure the IDL is always up to date.